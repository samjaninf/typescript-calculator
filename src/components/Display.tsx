import { StyleAttribute, css } from 'glamor'
import { h } from 'preact'

type DisplayProps = {
  output: string,
}

const display: StyleAttribute = css({
  alignItems: 'center',
  color: '#fafafa',
  display: 'flex',
  fontSize: '1.5em',
  height: '2.5em',
  justifyContent: 'flex-end',
  padding: '0 1em',
})

const Display = ({ output }: DisplayProps): JSX.Element => (
  <div className={display.toString()}>
    {output}
  </div>
)

export default Display
