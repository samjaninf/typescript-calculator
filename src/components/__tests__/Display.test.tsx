import Display from '../Display'
import snapshot from './utils/snapshot'

describe('Display', () => {
  it('renders', () => snapshot(Display))
  it('renders output property', () => snapshot(Display, { output: '13' }))
})
