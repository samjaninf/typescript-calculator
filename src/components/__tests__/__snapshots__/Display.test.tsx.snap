// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders 1`] = `"<div class=\\"css-1c40jf6\\"></div>"`;

exports[`Display renders output property 1`] = `"<div class=\\"css-1c40jf6\\">13</div>"`;
